File: ReactFinal/backend/413Final/413Final/Data/Entertainer.cs
using System.ComponentModel.DataAnnotations;
using System.ComponentModel.DataAnnotations.Schema;

namespace _413Final.Data;

public class Entertainer
{
    [Key]
    public int EntertainerId { get; set; }
    [Required]
    public string EntStageName { get; set; }
    [Required]
    public string EntSSN { get; set; }
    [Required]
    public string EntStreetAddress { get; set; }
    [Required]
    public string EntCity { get; set; }
    [Required]
    public string EntState { get; set; }
    [Required]
    public string EntZipCode { get; set; }
    [Required]
    public string EntPhoneNumber { get; set; }

    public string? EntWebPage { get; set; }
    public string? EntEMailAddress { get; set; }
    
    [Required]
    public string DateEntered { get; set; }
    
    
    // Computed Properties - Not Stored in the DB
    [NotMapped]
    public int TimesBooked { get; set; }
    [NotMapped]
    public DateTime? LastBookingDate { get; set; }
    // END of Computed Properties
}

File: ReactFinal/frontend/src/types/Entertainer.ts
export interface Entertainer {
  entertainerId: number;
  entStageName: string;
  entSSN: string;
  entStreetAddress: string;
  entCity: string;
  entState: string;
  entZipCode: string;
  entPhoneNumber: string;

  entWebPage: string | null;
  entEMailAddress: string | null;

  dateEntered: string;

  timesBooked: number;
  lastBookingDate: string | null;
}
